private/model/api/example.go

```diff
@@ -0,0 +1,342 @@
+				if (ref.Shape.MemberRefs[name].Streaming || ref.Shape.MemberRefs[name].Shape.Streaming) && ref.Shape.Payload == name {
+					ret += fmt.Sprintf("%s: aws.ReadSeekCloser(bytes.NewBuffer([]byte(%q))),\n", memName, v)
+				} else {
+					ret += fmt.Sprintf("%s: []byte(%q),\n", memName, v)
+				}
+			} else {
+				ret += fmt.Sprintf("%s: aws.String(%q),\n", memName, v)
+			}
+		case map[string]interface{}:
+			if ref == nil {
+				ret += fmt.Sprintf(`%s: {
+				%s
+			},
+			`, memName, buildShape(nil, v, isMap))
+			} else {
+				if isMap {
+					t = ref.Shape.Type
+				} else {
+					t = ref.Shape.MemberRefs[name].Shape.Type
+				}
+				switch t {
+				case "structure":
+					shapeName := ""
+					if ref.Shape.MemberRefs[name] != nil {
+						shapeName = ref.Shape.MemberRefs[name].Shape.ShapeName
+					} else {
+						shapeName = ref.Shape.ShapeName
+					}
+					ret += fmt.Sprintf(`%s: &%s.%s{
+				%s
+			},
+			`, memName, ref.API.PackageName(), shapeName, buildShape(ref.Shape.MemberRefs[name], v, false))
+				case "map":
+					shapeType := getNestedType(ref.Shape.MemberRefs[name].Shape.ValueRef.Shape)
+					ret += fmt.Sprintf(`%s: map[string]%s{
+				%s
+			},
+			`, name, shapeType, buildShape(&ref.Shape.MemberRefs[name].Shape.ValueRef, v, true))
+				}
+			}
+		}
+	}
+	return ret
+}
+
+// AttachExamples will create a new ExamplesDefinition from the examples file
+// and reference the API object.
+func (a *API) AttachExamples(filename string) {
+	p := ExamplesDefinition{API: a}
+
+	f, err := os.Open(filename)
+	defer f.Close()
+	if err != nil {
+		panic(err)
+	}
+	err = json.NewDecoder(f).Decode(&p)
+	if err != nil {
+		panic(err)
+	}
+
+	p.setup()
+}
+
+func (p *ExamplesDefinition) setup() {
+	keys := p.Examples.Names()
+	for _, n := range keys {
+		examples := p.Examples[n]
+		for i, e := range examples {
+			n = p.ExportableName(n)
+			e.OperationName = n
+			e.API = p.API
+			e.VisitedErrors = map[string]struct{}{}
+			op := p.API.Operations[e.OperationName]
+			e.OperationName = p.ExportableName(e.OperationName)
+			e.Operation = op
+			p.Examples[n][i] = e
+		}
+	}
+
+	p.API.Examples = p.Examples
+}
+
+// ExamplesGoCode will return a code representation of the entry within the
+// examples.json file.
+func (a *API) ExamplesGoCode() string {
+	var buf bytes.Buffer
+	imports := []string{
+		"fmt",
+		"bytes",
+		"time",
+		"github.com/aws/aws-sdk-go/aws",
+		"github.com/aws/aws-sdk-go/aws/session",
+		"github.com/aws/aws-sdk-go/aws/awserr",
+		"github.com/aws/aws-sdk-go/service/" + a.PackageName(),
+	}
+	buf.WriteString("import (\n")
+	for _, importValue := range imports {
+		buf.WriteString(fmt.Sprintf("%q\n", importValue))
+	}
+	buf.WriteString(")\n")
+	buf.WriteString("\nvar _ time.Duration\n")
+	buf.WriteString("var _ bytes.Buffer\n")
+	buf.WriteString("var _ aws.Config\n\n")
+	buf.WriteString(`func parseTime(layout, value string) *time.Time {
```

Comment:
All of these WriteString calls should write a single const string or from a template. Doing each line a separate calls will make it really easy to introduce errors.